File: astro1138/euler10.hs
eliminate' _ [] _ = []
eliminate' delta (e:list) n | e < n = e:(eliminate' delta list n)
                            | e == n = eliminate' delta list n
                            | e > n = eliminate' delta (e:list) (n + delta)
eliminate delta list = eliminate' delta list delta

prime_sieve [] = []
prime_sieve (n:rest) = n:rest''
    where rest' = eliminate n rest
          rest'' = prime_sieve rest'

euler_10 = (primes, s)
    where s = sum primes
          primes = prime_sieve [2..2000000]

main = do
  putStrLn $ show $ euler_10

File: astro1138/euler10.c
#define LIMIT 2000000

int main()
{
  char primes[LIMIT];
  int i, n = 0;
  register long long sum = 0;

  for(i = 2; i < LIMIT; ++i)
  {
    primes[i] = 1;
  }
  for(i = 2; i < LIMIT; ++i)
  {
    if (primes[i])
    {
      int j;
      if (n % 100 == 0)
        printf("%i\n", i);
      n++;
      sum += i;
      for(j = i * 2; j < LIMIT; j += i)
      {
        if (j % i == 0)
          primes[j] = 0;
      }
    }
  }
  printf("Sum: %lli\n", sum);
}
